test.js

```diff
@@ -6,100 +6,146 @@ const crypto = require('crypto')
       setTimeout(() => {
-        fastify.kafka.push({
-          topic: 'test',
+        producerFastify.kafka.push({
+          topic: topicName,
           payload: 'hello world!',
           key: 'testKey'
         })
+
+        t.ok(true)
       }, 1000)
     })
 })
 
-test('Register to multiple topics', t => {
-  t.plan(3)
-
-  const fastify = Fastify()
-  const group = crypto.randomBytes(20).toString('hex')
-  options.producer['group.id'] = group
+test('multiple topics', t => {
+  t.plan(7)
+  const options = copyPlainObject(defaultOptions)
+  const group = generateGroupId()
   options.consumer['group.id'] = group
+  options.producer['group.id'] = group
+
+  const topicName1 = generateTopicName()
+  const topicName2 = generateTopicName()
 
-  fastify
-    .register(fastifyKafka, options)
+  const producerFastify = Fastify({ logger })
+  const consumerFastify = Fastify({ logger })
+
+  t.tearDown(() => producerFastify.close())
+  t.tearDown(() => consumerFastify.close())
+
+  consumerFastify
+    .register(fastifyKafka, {...options, producer: undefined})
     .after(err => {
       t.error(err)
 
-      fastify.kafka.producer.on('error', t.fail)
-      fastify.kafka.consumer.on('error', t.fail)
-
-      fastify.kafka.subscribe(['test', 'kafka'])
+      consumerFastify.kafka.consumer.on('error', t.fail)
+      consumerFastify.kafka.subscribe([topicName1, topicName2])
 
-      fastify.kafka.on('test', (msg, commit) => {
-        t.strictEqual(msg.value.toString(), 'hello world!')
+      consumerFastify.kafka.on(topicName1, (msg, commit) => {
+        t.strictEqual(msg.value.toString(), 'topic1')
         commit()
-
-        fastify.kafka.push({
-          topic: 'kafka',
-          payload: 'winter is coming',
-          key: 'testKey'
-        })
+        t.ok(true)
       })
 
-      fastify.kafka.on('kafka', (msg, commit) => {
-        t.strictEqual(msg.value.toString(), 'winter is coming')
+      consumerFastify.kafka.on(topicName2, (msg, commit) => {
+        t.strictEqual(msg.value.toString(), 'topic2')
         commit()
-        fastify.close()
+        t.ok(true)
       })
 
-      fastify.kafka.consume()
+      consumerFastify.kafka.consume()
+    })
+
+  producerFastify
+    .register(fastifyKafka, {...options, consumer: undefined})
+    .after(err => {
+      t.error(err)
 
+      producerFastify.kafka.producer.on('error', t.fail)
       setTimeout(() => {
-        fastify.kafka.push({
-          topic: 'test',
-          payload: 'hello world!',
+        producerFastify.kafka.push({
+          topic: topicName1,
+          payload: 'topic1',
           key: 'testKey'
         })
+        producerFastify.kafka.push({
+          topic: topicName2,
+          payload: 'topic2',
+          key: 'kafkaKey'
+        })
+
+        t.ok(true)
       }, 1000)
     })
 })
+
+function generateGroupId () {
+  return crypto.randomBytes(20).toString('hex')
+}
+
+function copyPlainObject (obj) {
+  return JSON.parse(JSON.stringify(obj))
+}
```

Comment:
Can you add a comment that states it is really SLOW and to not use it outside of tests.

Reply:
done